src/core/utils/channel_manager.hpp

```diff
@@ -91,16 +111,32 @@ class ChannelManager : public InstanceLocator, private NonCopyable
      * @param[in] aChannel             The new channel for the Thread network.
      *
      */
-    void RequestChannelChange(uint8_t aChannel);
+    void RequestNetworkChannelChange(uint8_t aChannel);
+#endif
 
+#if OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE
     /**
-     * Gets the channel from the last successful call to `RequestChannelChange()`.
+     * Changes the csl channel.
+     *
+     * The csl channel changes to the given channel only if csl is active.
+     *
+     * If the requested channel changes, it will trigger a `Notifier` event `kEventChannelManagerNewChannelChanged`.
+     *
+     * @param[in] aChannel             The new channel for the csl link.
+     *
+     */
+    void ChangeCslChannel(uint8_t aChannel);
+#endif
+
+    /**
+     * Gets the channel from the last successful call to `RequestChannelChange()` or `ChangeCslChannel()`.
```

Comment:
Change this back?

Reply:
It is still correct. I think it if helpful to have this insight into channel manager's last selection.